build.zig

```diff
@@ -149,6 +152,8 @@ pub fn build(b: *std.Build) !void {
         posixLib_mod.addCMacro("PCRE2_CODE_UNIT_WIDTH", @tagName(codeUnitWidth));
         if (linkage == .static) {
             posixLib_mod.addCMacro("PCRE2_STATIC", "");
+        } else {
+            posixLib_mod.addCMacro("PCRE2POSIX_DYNAMIC", "");
```

Comment:
```suggestion
            posixLib_mod.addCMacro("PCRE2POSIX_SHARED", "");
            pcre2test_mod.addCMacro("PCRE2POSIX_SHARED", "");
```

PCRE2POSIX_DYNAMIC doesn't exist. I think you meant SHARED.

Also, it should be added to pcre2test_mod as well.